### PR TITLE
CLI: Backup: SharePoint:  Pages Enable 

### LINKED ISSUE #2107: CLI: SharePoint: --data command ineffective
The following command:
`$ ./corso backup create sharepoint --site "<siteID>" --data libraries` should ensure that libraries only are returned. Lists and Libraries are created with this backup. Additionally, if `library` is used, the backup is still run rather than returning an error.

### LINKED ISSUE #2216: CLI: Backup: SharePoint: Enable Pages
Enable selection of `SharePoint.Pages` in Corso for testing.

### PR DESCRIPTION
**Pull request by dadams39**
## Description
Adds the functionality to select SharePoint Pages for backup.

## Does this PR need a docs update or release note?
SharePoint commands remain hidden in terms of pages. 
- [x] :clock1: Yes, but in a later PR

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* closes  #2216 <issue>
* closes #2107

## Test Plan

- [x] :zap: Unit test

### CODE REVIEW

**Comment by dadams39** [comment]
On local branch:
```bash 
./corso backup create sharepoint --site <siteID> --data pages
Connecting to repository:      6s done 
Connecting to M365:      0s done 
Discovering items to backup:      0s done 
∙ pages - <siteId>:         done 
Backing up data:      0s done 
  Started At            ID                                    Status              Resource Owner                                                                                 
  2023-01-21T00:01:08Z  <backupID>     No Data (0 errors)  <siteID>
  ```

File: src/cli/backup/sharepoint.go [hunk 1]
```diff
@@ -128,11 +132,22 @@ func addSharePointCommands(cmd *cobra.Command) *cobra.Command {
...
+
+		fs.StringSliceVar(
+			&pageFolders,
+			utils.PageFN, nil,
+			"Select backup data by site ID; accepts '"+utils.Wildcard+"' to select all sites.")
+
+		fs.StringSliceVar(
+			&page,
+			utils.PageItemFN, nil,
+			"Select backup data by site page ID; accepts '"+utils.Wildcard+"' to select all pages within site.",
```
Comment (ryanfkeepers) [inline src/cli/backup/sharepoint.go:149]:
```suggestion
			"Select backup data by site page ID; accepts '"+utils.Wildcard+"' to select all pages within the site.",
```

File: src/cli/backup/sharepoint.go [hunk 2]
```diff
@@ -260,13 +275,21 @@ func validateSharePointBackupCreateFlags(sites, weburls []string) error {
 		)
 	}
 
+	for _, d := range data {
+		if d != dataLibraries && d != dataPages {
+			return errors.New(
+				d + "is an unrecognized data type; either  " + dataLibraries + "or " + dataPages,
```
Comment (ryanfkeepers) [inline src/cli/backup/sharepoint.go:281]:
```suggestion
				d + " is an unrecognized data type; must be either  " + dataLibraries + "or " + dataPages,
```

File: src/cli/backup/sharepoint.go [hunk 3]
```diff
@@ -297,7 +320,20 @@ func sharePointBackupCreateSelectors(
 	}
 
 	sel := selectors.NewSharePointBackup(union)
-	sel.Include(sel.AllData())
+	if len(data) == 0 {
+		sel.Include(sel.AllData())
+
```
Comment (ryanfkeepers) [inline src/cli/backup/sharepoint.go:325]:
```suggestion
```

File: src/cli/backup/sharepoint.go (hunk 1 above)
Comment (ryanfkeepers) [inline src/cli/backup/sharepoint.go:149]: we've been picky about forcing users to use IDs in the past.  Do sites have filenames that might make for a better use interface?

**Review by ryanfkeepers — APPROVED** [review]

File: src/cli/backup/sharepoint.go (hunk 1 above)
Comment (dadams39) [reply src/cli/backup/sharepoint.go:149]: You are correct. This isn't a great use case. Will switch to full page name (e.g. `home.aspx`).

Comment (dadams39) [reply src/cli/backup/sharepoint.go]: Resolved within [bc5e8ca](https://github.com/alcionai/corso/pull/2213/commits/bc5e8ca5a64ff5cbe622e7cb7ef44aacdd245c79)

File: src/cli/backup/sharepoint.go (hunk 2 above)
Comment (dadams39) [reply src/cli/backup/sharepoint.go]: Resolved within [bc5e8ca](https://github.com/alcionai/corso/pull/2213/commits/bc5e8ca5a64ff5cbe622e7cb7ef44aacdd245c79)

File: src/cli/backup/sharepoint.go (hunk 3 above)
Comment (dadams39) [reply src/cli/backup/sharepoint.go:325]: Space needed for GOFMT: wsl

File: src/cli/backup/sharepoint.go (hunk 1 above)
Comment (dadams39) [reply src/cli/backup/sharepoint.go]: Resolved [8d70659](https://github.com/alcionai/corso/pull/2213/commits/8d70659e1e547e6e72dc8d58a618de112d6ba5d8)

**Pull request merged**